src/test/java/com/qunite/api/web/AuthenticationIntegrationTest.java

```diff
@@ -0,0 +1,231 @@
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.testcontainers.shaded.com.fasterxml.jackson.databind.ObjectMapper;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@AutoConfigureMockMvc
+@IntegrationTest
+class AuthenticationIntegrationTest {
+
+  private final String url = "auth";
+
+  @Autowired
+  private MockMvc mockMvc;
+
+  @Autowired
+  private UserMapper userMapper;
+
+  @Autowired
+  private UserService userService;
+
+  @Autowired
+  private UserRepository userRepository;
+
+  @AfterEach
+  public void cleanAll() {
+    userRepository.deleteAll();
+  }
+
+  @Test
+  void unauthenticatedUserShouldAccessEndpoints() throws Exception {
+    mockMvc.perform(post("/{url}/sign-up", url))
+        .andExpect(status().isBadRequest());
+    mockMvc.perform(post("/{url}/sign-in", url))
+        .andExpect(status().isBadRequest());
+  }
+
+  @Test
+  void signUpShouldCreateNewUserWithValidData() throws Exception {
+    var user = new User();
+    user.setUsername("John");
+    user.setPassword("Johnson");
+    user.setEmail("john@gmail.com");
+
+    var dto = userMapper.toUserCreationDto(user);
+    var json = new ObjectMapper().writeValueAsString(dto);
+
+    mockMvc.perform(
+            post("/{url}/sign-up", url)
+                .contentType(MediaType.APPLICATION_JSON).content(json))
+        .andExpect(status().is2xxSuccessful());
+
+    assertThat(userService.findByUsername("John")).isPresent();
+  }
+
+  @Test
+  @Sql("/users-create.sql")
+  void signUpShouldNotCreateWithExistingUsername() throws Exception {
+    var user = new User();
+    user.setUsername("First");
+    user.setPassword("Johnson");
+    user.setEmail("john@gmail.com");
+
+    var dto = userMapper.toUserCreationDto(user);
+    var json = new ObjectMapper().writeValueAsString(dto);
+
+    mockMvc.perform(
+            post("/{url}/sign-up", url)
+                .contentType(MediaType.APPLICATION_JSON).content(json))
+        .andExpect(status().isBadRequest());
+  }
+
+  @Test
+  @Sql("/users-create.sql")
+  void signUpShouldNotCreateWithExistingEmail() throws Exception {
+    var user = new User();
+    user.setUsername("John");
+    user.setPassword("Johnson");
+    user.setEmail("User1@user.com");
+
+    var dto = userMapper.toUserCreationDto(user);
+    var json = new ObjectMapper().writeValueAsString(dto);
+
+    mockMvc.perform(
+            post("/{url}/sign-up", url)
+                .contentType(MediaType.APPLICATION_JSON).content(json))
+        .andExpect(status().isBadRequest());
+  }
+
+  @Test
+  void signUpShouldEncryptPassword() throws Exception {
+    var user = new User();
+    user.setUsername("John");
```

Comment:
A lot of variables can be extracted to class field. Just like `private static final DUMMY_USERNAME = "John"`

Reply:
Agree, but not sure what to do with things like this(occured after @vetterocki  requests)
![image](https://github.com/q-unite/qunite-api/assets/103951082/9c1a3b07-7f98-495c-b84d-d04e31552d4c)